src/features/MovieList/popularMoviesSaga.js

```diff
@@ -0,0 +1,22 @@
+import { call, delay, put, takeLatest } from "redux-saga/effects";
+import { popularMovies } from "./popularMovies";
+import {
+  fetchPopularMovies,
+  fetchPopularMoviesError,
+  fetchPopularMoviesSuccess,
+} from "./popularMoviesSlice";
+
+function* fetchPopularMoviesHandler() {
+  try {
+    const popularMoviesList = yield call(popularMovies);
+    yield delay(1000);
+    yield put(fetchPopularMoviesSuccess(popularMoviesList));
+  } catch (error) {
+    yield put(fetchPopularMoviesError());
+    yield call(alert, "Download failed");
+  }
+}
+
+export function* popularMoviesSaga() {
+  yield takeLatest(fetchPopularMovies, fetchPopularMoviesHandler);
```

Comment:
I think we are missing .type in this function
`export function* popularMoviesSaga() {
  yield takeLatest(fetchPopularMovies.type, fetchPopularMoviesHandler);

Add a suggestion, <Ctrl+g>
Add heading textAdd bold text, <Ctrl+b>Add italic text, <Ctrl+i>
Add a quote, <Ctrl+Shift+.>Add code, <Ctrl+e>Add a link, <Ctrl+k>
`